### DISCUSSION TITLE
Could @babel/runtime be used as a dev dependency in bundle?

### DISCUSSION THREAD

**Discussion by PaperStrike** [babel/babel#13569] [Q&A]
Babel in some projects is only used when bundling, e.g., webpack with babel-loader. The distributions are some standalones. In that case, would it be better to use @babel/runtime as a [*dev dependency*](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#devdependencies) instead of a [*dependency*](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#dependencies)?

**Comment by JLHwung** [comment]
No. Unlike most Babel packages, `@babel/runtime` is supposed to be shipped with your artifacts. See also https://babeljs.io/docs/en/babel-runtime#usage

**Reply by PaperStrike** [reply]
Thank you!
Well but, I have read the document, and that's why I ask this question. To my understanding, the reason `@babel/runtime` should be a *dependency* is that the Babel output needs this package to run. However, in the case I suggested, `@babel/runtime` is *bundled with the output*. Considering users already have this package in the bundles, what could benefit from requiring users to install another outside?
(There's one exactly in this case, [@sliphua/pjax](https://www.npmjs.com/package/@sliphua/pjax).)

---

That's deep but I finally dug out something supporting my point: https://github.com/babel/website/pull/424. The runtime package was particularly suggested to be installed as a _dev dependency_ when using with webpack, in a section that seems to have been ignored since https://github.com/babel/babel/commit/597d895cf3b22b2dd6404534ae8335c3de42b30f (for conciseness?).

**Reply by nicolo-ribaudo** [reply]
`@babel/runtime` should be put in the same dependencies where you would put React, jQuery, lodash, etc.

If you are bundling them, then yes it would be fine.

**Reply by PaperStrike** [reply]
@nicolo-ribaudo Thank you!
Then the first "NOTE" in https://github.com/babel/babel-loader#babel-is-injecting-helpers-into-each-file-and-bloating-my-code seems somehow misleading. Considering `babel-loader` is always used in bundling, and `@babel/runtime` is almost always being bundled to run in browsers, changes to the "must" statement seem necessary.

---

I am not bundling and will never bundle React, jQuery, nor lodash. They are designed to run seperately as a single script and are large. `@babel/runtime` is modular and small. It's more like polyfills, right?
